cmusatyalab/openface
# Unknown people benchmarking with lfw dataset.

### What does this PR do?
Improved the [previous PR](https://github.com/cmusatyalab/openface/pull/165) on unknown classification benchmark.

### Where should the reviewer start?
[./evaluation/lfw-classification-unknown.py](https://github.com/Vijayenthiran/openface/blob/classification-unknown/evaluation/lfw-classification-unknown.py)

### How should this PR be tested?

1. _Preprocess the lfw dataset:_

 `./evaluation/lfw-classification-unknown.py --verbose preprocess --lfwDir <lfw-directory-location> --rangeOfPeople <range-of-people-to-be-taken-as-known-dataset> --featuresDir <directory-where-features-should-be-stored>`

 Eg:
 .`/evaluation/lfw-classification-unknown.py --verbose preprocess --lfwDir /Users/Vijay/scikit_learn_data/lfw_home/lfw_funneled --rangeOfPeople 0:10 --featuresDir /Users/Vijay/Documents/ImageProcessing/FaceRecognition/openface/data/lfw_benchmark`


2. _Train the dataset and create different classifiers:_

 `./evaluation/lfw-classification-unknown.py train <feature-directory>`

 Eg:
 `./evaluation/lfw-classification-unknown.py train ./data/lfw_benchmark/train_known_features `

3. _Run the classification on known dataset:_

 `./evaluation/lfw-classification-unknown.py inferFromTest <feature-directory> <known-test-dataset>`

 <known-test-dataset> is generated from the preprocess step.

 Eg:
 `./evaluation/lfw-classification-unknown.py inferFromTest ./data/lfw_benchmark/train_known_features ./data/lfw_benchmark/test_known_raw`


4. _Run the classification on unknown dataset:_

 `./evaluation/lfw-classification-unknown.py inferFromTest <feature-directory> <unknown-test-dataset> --threshold <threshold-value-in-float> --unknown`

 <unknown-test-dataset> is generated from preprocess step.

 Eg:
 `./evaluation/lfw-classification-unknown.py inferFromTest ./data/lfw_benchmark/train_known_features ./data/lfw_benchmark/test_unknown_raw --threshold 0.9 --unknown`

### Any background context you want to provide?
The PR helps to create a benchmark of unknown people classification on lfw dataset.

### What are the relevant issues?

#144 

# Screenshots (if appropriate)

# Questions:

+ Do the docs need to be updated?<br>
Yes, As per your [previous comment](https://github.com/cmusatyalab/openface/pull/165#issuecomment-234227383)
+ Does this PR add new (Python) dependencies?<br>
Yes. Updated in requirement.txt
